dropwizard-auth/src/test/java/io/dropwizard/auth/CachingAuthenticatorTest.java

```diff
@@ -104,4 +105,16 @@ public void calculatesCacheStats() throws Exception {
         cached.authenticate("credentials1");
         assertThat(cached.stats().loadCount()).isEqualTo(1);
     }
+
+    @Test
+    public void shouldNotCacheAbsentPrincipals() throws Exception {
+        when(underlying.authenticate(anyString())).thenReturn(Optional.<Principal>absent());
+        try {
+            cached.authenticate("credentials");
+            failBecauseExceptionWasNotThrown(AuthenticationException.class);
```

Comment:
Invalid credentials should in fact never cause a `AuthenticationException`. That exception should only be used if the backend is unavailable and the credentials can't be verified.